### PR TITLE
Bump jupyterhub chart and hub image to version 1.1.3

### LINKED ISSUE #539: Upgrade to z2jh 1.1.0+
# Summary

Task: to upgrade to z2jh 1.1.0 ([CHANGELOG](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/main/CHANGELOG.md))

- hub.services.<service>.apiToken is now automatically generated, and a bug in jupyterhub is fixed so it can be set to a blank even if it wouldn't be autogenerated by z2jh. I recall @GeorgianaElena ran into this bug before and perhaps there is some cleanup that can be done.
- I believe we install a specific commit of kubespwner that previously wasn't released in the pilot-hubs hub image, but that won't be needed any more
- KubeSpawner now intelligently sort environment variables to help them resolve. This can make our logic to set SCRATCH_BUCKET environment variables less complicated if i recall correctly.

### PR DESCRIPTION
**Pull request by sgibson91**
This brings in authenticating with GitHub Teams feature from
OAuthenticator

first step towards #625

EDIT: closes #539

### CODE REVIEW

**Review by yuvipanda — CHANGES_REQUESTED** [review]
I think you need to run `chartpress --push` to build and push the new hub image with the new base.

**Comment by sgibson91** [comment]
Happy to let someone else do that if it can't wait until I'm back? (About to go pick up my cat-sitter!) I don't think this is super urgent as we're not working on the OAuthenticator stuff yet :)

**Comment by yuvipanda** [comment]
@sgibson91 absolutely! PLEASE GO VACATION! ROME NEEDS YOU!!!!!! SEND PICS :)

**Comment by choldgraf** [comment]
**Update**: We plan to merge this in ~12 hours (@GeorgianaElena will shepherd and others should be ready to assist if something unexpected happens!)

File: hub-templates/basehub/Chart.yaml
Comment (GeorgianaElena) [inline hub-templates/basehub/Chart.yaml:5]:
```suggestion
version: "0.1.0"
```

**Comment by GeorgianaElena** [comment]
Before merging this, I tried to deploy it first manually only on the staging hub.
But, I bumped into this chart version mismatch when trying to do so. We declare a new version for the `basehub` chart, but the `daskhub` chart depends still on the old one.

My guess is that the `basehub` chart version was changed by mistake. @consideRatio, is this right or am I missing something?
Should we instead (of my code suggestion) have the `daskhub` chart depend on the new basehub chart version (`version: "0.0.1-n1159.h5b045cd"`)?

**Comment by consideRatio** [comment]
Oh doh, great catch @GeorgianaElena! Either make daskhub reference basehubs new version, or make basehubs version be 0.1.0 again or whatever it was before.

Also update the inline comment about using chartpress about the need to keep basehubs chart version in sync with the dependency reference in daskhub.

**Comment by GeorgianaElena** [comment]
Thanks @consideRatio! The staging deploy went ok and I was able to login and start a server on that hub afterwards.
So, I'll go ahead and push the merge button now :crossed_fingers:

Thanks everyone! :sun_with_face:

**Comment by sgibson91** [comment]
Thank you for stewarding @GeorgianaElena and @consideRatio for assisting! 🎉

**Pull request merged**
